src/irmin-pack/unix/gc.ml

```diff
@@ -543,72 +881,72 @@ module Make (Args : Args) = struct
-            @@ fun () -> read_gc_output ~root:t.root ~generation:t.generation
+            read_gc_output ~root:t.root ~generation:t.generation
           in
 
           let result =
             let open Result_syntax in
             match (status, gc_output) with
-            | `Success, Ok copy_end_offset ->
+            | `Success, Ok worker_stats ->
+                let new_suffix_end_offset =
+                  match
+                    List.assoc_opt "suffix" worker_stats.Stats.Latest_gc.files
+                  with
+                  | Some x -> x
+                  | None -> assert false
+                in
+                let stats_builder =
+                  Main_stats.finish_current_step stats_builder
+                    "copy latest newies"
+                in
                 let* new_suffix_end_offset =
-                  time (fun t ->
-                      s := { !s with transfer_latest_newies_duration = t })
-                  @@ fun () ->
                   transfer_latest_newies ~right_start_offset:t.offset
-                    ~copy_end_offset t
+                    ~copy_end_offset:new_suffix_end_offset t
+                in
+                let stats_builder =
+                  Main_stats.finish_current_step stats_builder "swap and purge"
                 in
                 let* () =
-                  time (fun t -> s := { !s with swap_duration = t })
-                  @@ fun () ->
                   swap_and_purge ~right_start_offset:t.offset
                     ~right_end_offset:new_suffix_end_offset t
                 in
-                (if t.unlink then
-                 time (fun t -> s := { !s with unlink_duration = t })
-                 @@ fun () -> unlink_all t);
-
-                let duration = t.elapsed () in
-                s :=
-                  {
-                    !s with
-                    duration;
-                    finalisation_duration = duration -. start;
-                  };
-                t.stats <- Some !s;
+                let stats_builder =
+                  Main_stats.finish_current_step stats_builder "unlink"
+                in
+                if t.unlink then unlink_all t;
+
+                let stats_builder =
+                  let after_suffix_end_offset =
+                    Dispatcher.end_offset t.dispatcher
+                  in
+                  Main_stats.finalise stats_builder worker_stats
+                    ~after_suffix_end_offset
+                in
+                t.resulting_stats <- Some stats_builder;
 
                 [%log.debug
-                  "Gc ended. %a, newies bytes:%a" pp_stats !s Int63.pp
-                    (Int63.sub new_suffix_end_offset copy_end_offset)];
-                let () = Lwt.wakeup_later t.resolver (Ok !s) in
-                Ok (`Finalised !s)
+                  "Gc ended successfully. %a"
+                    (Irmin.Type.pp Stats.Latest_gc.stats_t)
```

Comment:
I'm concerned that this log line is a large output and too much information. Related to my plea to keep the simplified stats, I think it is better to log a minimal amount here (like we are currently doing).

Reply:
Didn't notice that it's debug. Agreed.